packages/x-codemod/src/v8.0.0/charts/rename-sparkline-colors-to-color/expected.spec.tsx

```diff
@@ -0,0 +1,19 @@
+/* eslint-disable no-restricted-imports */
+import * as React from 'react';
+import { SparkLineChart } from '@mui/x-charts';
+
+const data = [1, 2];
+
+function Chart() {
+  const fn = (mode) => (mode === 'light' ? ['black'] : ['white']);
+
+  // prettier-ignore
+  return (
+    (<React.Fragment>
+      <SparkLineChart data={data} color={['red'][0]} />
+      <SparkLineChart data={data} color={typeof fn === "function" ? mode => fn(mode)?.[0] : fn} />
+      <SparkLineChart data={data} color={mode => (mode => (mode === 'light' ? ['black'] : ['white']))(mode)?.[0]} />
```

Comment:
This might seem a bit confusing, but what I'm doing is: 

Given an arrow function declaration `fnDecl`, transform it into: `mode => (fnDecl)(mode)?.[0]`.

We need the optional chaining operator because `undefined` is a valid value for `colors`, so we need be careful not to access `[0]` of an undefined value.

Reply:
It's tricky and I think most of the users will prefer to just rewrite the function by themself.